### PR TITLE
Bastion logging

### LINKED ISSUE #60: Add logging for bastion when accessing it via ssh over ssm
When utilizing ssh or sshuttle over SSM, logs are not currently being captured. 

- Identify minimum logs needed
- ensure those logs are being shipped to the appropriate logging bucket

### PR DESCRIPTION
**Pull request by JaseKoonce**
Added to user data to download and configure its config file. Set-up logging of commands and access time.

### CODE REVIEW

File: modules/bastion/ssm.tf [hunk 1]
```diff
@@ -78,3 +90,160 @@ resource "aws_ssm_document" "session_manager_prefs" {
...
+
+  event_pattern = <<EOF
+  {
+  "source": ["aws.ssm"],
+  "detail-type": ["AWS API Call via CloudTrail"],
+  "detail": {
+  "eventSource": ["ssm.amazonaws.com"],
+  "eventName": ["IAMUser","StartSession"]
+  }
+}
```
Comment (wirewc) [inline modules/bastion/ssm.tf:241]: This json is not valid, I think it has too many closing curly braces.

Comment (JaseKoonce) [reply modules/bastion/ssm.tf:241]: Just went through it, I didn't notice any problems. All curly brackets are accounted for. Happy to look over again if you see anything.

**Comment by JaseKoonce** [comment]
This branch adds the following features:
- limits concurrent ssh sessions to one for more reliable ssh logging
- Sets idle timeout to 4 hours
- adds more detailed logging for both ssh and ssm through the cloudwatch agent and cloudtrails

**Comment by ntwkninja** [comment]
image

File: modules/bastion/iam.tf [hunk 2]
```diff
@@ -393,3 +394,57 @@ resource "aws_iam_policy" "terraform_policy" {
...
+# Create custom policy for KMS
+data "aws_iam_policy_document" "kms_access" {
+  # checkov:skip=CKV_AWS_111: todo reduce perms on key
+  # checkov:skip=CKV_AWS_109: ADD REASON
+  statement {
+    sid = "KMS Key Default"
+    principals {
+      type = "AWS"
+      identifiers = [
+        "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:root"
```
[the remark this answers is not in the document]
Comment (ntwkninja) [reply modules/bastion/iam.tf:406]: no change needed here

https://github.com/awsdocs/aws-kms-developer-guide/blob/master/doc_source/key-policy-default.md?plain=1#L18-L41

File: modules/bastion/ssm.tf (hunk 1 above)
Comment (ntwkninja) [reply modules/bastion/ssm.tf]: https://github.com/defenseunicorns/iac/pull/66/files#diff-aafdffe7f6b627a90c2f71120869c58a3f693ee17a108c27d359b1e118804223R32-R42

**Review by ntwkninja — APPROVED** [review]
🚀

**Pull request merged**